core/types/transaction.go

```diff
@@ -589,6 +589,35 @@ func NewTransactionsByPriceAndNonce(signer Signer, txs map[common.AddressBytes]T
 	}
 }
 
+// NewTransactionsByPriceAndNonce creates a transaction set that can retrieve
+// transactions in a nonce-honouring way. Also, sender is not checked because the
+// txpool has already done this.
+//
+// Note, the input map is reowned so the caller should not interact any more with
+// if after providing it to the constructor.
+func NewTransactionsByNonceNoSorting(signer Signer, txs map[common.AddressBytes]Transactions, baseFee *big.Int) *TransactionsByPriceAndNonce {
```

Comment:
I would prefer this was just a "sort" flag passed into `NewTransactionsByPriceAndNonce()` rather than a whole new function.

Reply:
Fixed